contracts/RelayerHub.sol

```diff
@@ -8,88 +8,189 @@ import "./System.sol";
+
+        } else {
+            require(false, "unknown param");
+        }
+        emit paramChange(key, value);
+    }
+
+    function removeManagerByGov(address payable managerToBeRemoved) internal {
+        removeManagerHelper(managerToBeRemoved);
+    }
+
+    function removeManager() external onlyRegisteredManager {
+        // here the manager removes himself
+        removeManagerHelper(payable(msg.sender));
+    }
+
+    function removeManagerHelper(address payable managerAddress) internal {
+        // check if the manager address already exists
+        require(relayManagersExistMap[managerAddress], "manager doesn't exist");
+
+        address relayerAddress = managersAndRelayers[managerAddress];
+
+        delete (relayManagersExistMap[managerAddress]);
+        delete (managersAndRelayers[managerAddress]);
+
+        manager memory a = managers[managerAddress];
+        managerAddress.transfer(a.deposit.sub(a.dues));
+        address payable systemPayable = payable(address(uint160(SYSTEM_REWARD_ADDR)));
+        systemPayable.transfer(a.dues);
+
+        delete (managers[managerAddress]);
+        delete (managersRegistered[managerAddress]);
+
+        // emit success event
+        emit removeManagerByGovEvent(managerAddress);
+        if (relayerAddress != address(0)) {
+            emit removeRelayerEvent(relayerAddress);
+        }
+    }
+
+    function addManagerByGov(address managerToBeAdded) internal {
+        require(!relayManagersExistMap[managerToBeAdded], "manager already exists");
+        require(!isContract(managerToBeAdded), "contract is not allowed to be a manager");
+
+        relayManagersExistMap[managerToBeAdded] = true;
+
+        emit addManagerByGovEvent(managerToBeAdded);
+    }
+
+    function registerManager() internal onlyNonRegisteredManager {
+        require(msg.value == requiredDeposit, "deposit value is not exactly the same");
+        managers[msg.sender] = manager(requiredDeposit, dues);
+        managersRegistered[msg.sender] = true;
+        emit registerManagerEvent(msg.sender);
+    }
+
+    function addRelayer(address relayerToBeAdded) public onlyRegisteredManager noProxy {
+        require(!relayerExistsMap[relayerToBeAdded], "relayer already exists");
+        require(!isContract(relayerToBeAdded), "contract is not allowed to be a relayer");
+
+        if(managersAndRelayers[msg.sender] != address(0)) {
+            address r = managersAndRelayers[msg.sender];
+            delete (relayerExistsMap[r]);
+            emit removeRelayerEvent(r);
+        } 
+        
+        managersAndRelayers[msg.sender] = relayerToBeAdded;
+        relayerExistsMap[relayerToBeAdded] = true;
+        emit addRelayerEvent(relayerToBeAdded);
+    }
+
+    function registerManagerAddRelayer(address relayer) external payable onlyNonRegisteredManager {
+        registerManager();
+        addRelayer(relayer);
+    }
+
+    function removeRelayer() external onlyRegisteredManager {
```

Comment:
I don't think there is a necessity to implement `removeRelayer`, we can do it by updating the relayer to a non-exist account.